File: C++17/백준/Bronze/29163. Счастье Мистера Бина/Счастье Мистера Бина.cc
#include <iostream>
#include <vector>
using namespace std;

int main (void) {
    int N;
    cin >> N;
    
    int even_cnt = 0;
    int odd_cnt = 0;
    
    for (int i = 0; i < N; i++) {
        int num;
        cin >> num;
        
        if (num % 2 == 0) {
            even_cnt++;
        } else {
            odd_cnt++;
        }
    }
    
    if (even_cnt > odd_cnt) {
        cout << "Happy";
    } else {
        cout << "Sad";
    }
    
    return 0;
}

File: Java/백준/Bronze/29163. Счастье Мистера Бина/Счастье Мистера Бина.java
import java.util.Scanner;

public class Main {
    public static void main (String[] args) {
        Scanner sc = new Scanner (System.in);
        
        int n = sc.nextInt();
        
        int even_cnt = 0;
        int odd_cnt = 0;
        
        for (int i = 0; i < n; i++) {
            long num = sc.nextLong();
            
            if (num % 2 == 0) {
                even_cnt++;
            } else {
                odd_cnt++;
            }
        }
        
        if (even_cnt > odd_cnt) {
            System.out.print("Happy");
        } else {
            System.out.print("Sad");
        }
        
        sc.close();
    }
}